packages/bits/src/lib/wizard-v2/wizard-horizontal/wizard-horizontal.component.ts

```diff
@@ -41,10 +80,236 @@ export class WizardHorizontalComponent extends WizardDirective implements OnInit
+                            this.overflownStepsStart = this.overflownStepsStart.concat(this.visibleSteps.splice(0, selectedIndex + 1));
+                        }
+                    } else {
+                        if (this.overflownStepsStart.length) {
+                            const startPiece = this.overflownStepsStart.splice(event.selectedIndex, this.overflownStepsStart.length - event.selectedIndex);
+                            this.visibleSteps = startPiece.concat(this.visibleSteps);
+                            this.overflownStepsEnd.splice(0, 0, ...this.visibleSteps.splice(this.visibleSteps.length - startPiece.length, startPiece.length));
+                        }
+                    }
+                }
+            });
+
+        this.headerResizeObserver.observe(this.headerContainer.nativeElement);
+    }
+
+    public ngOnDestroy(): void {
+        super.ngOnDestroy();
+
+        this.headerResizeObserver.disconnect();
+    }
+
+    private handleOverflow(): void {
+        this.checkOverflow();
+
+        if (this.hasOverflow || this.isInResponsiveMode) {
+            const numberOfVisibleItems = this.numberOfVisibleItems;
+
+            if (!this.isInResponsiveMode) {
+                if (this.state?.finished) {
+                    this.visibleSteps = this.stepsArray.slice(-numberOfVisibleItems);
+                    this.overflownStepsStart = this.stepsArray.slice(0, -numberOfVisibleItems);
+
+                    // This covers an edge case with the saved wizard state. It can happen that the active element
+                    // appears inside the hidden list of items, when the dynamnic steps are re-added.
+                    // We make it visible again by retrieving it back to the visible items array
+                    if (this.overflownStepsStart.length && !this.isCurrentStepVisible) {
+                        do {
+                            this.takeLastAddFirst(this.overflownStepsStart, this.visibleSteps);
+                            this.takeLastAddFirst(this.visibleSteps, this.overflownStepsEnd);
+                        }
+                        while(!this.isCurrentStepVisible)
+                    }
+                } else {
+                    this.visibleSteps = this.stepsArray.slice(0, numberOfVisibleItems);
+                    this.overflownStepsEnd = this.stepsArray.slice(numberOfVisibleItems);
+                }
+            } else {
+                this.handleDynamicHeaderChanges(numberOfVisibleItems);
+            }
+        }
+
+        this.checkResponsiveMode();
+    }
+
+    private get isCurrentStepVisible(): boolean {
+        return this.visibleSteps.includes(this.stepsArray[this.selectedIndex]);
+    }
+
+    private get numberOfVisibleItems(): number {
+        return Math.ceil((this.headerContainerWidth - this.headerPaddings * 2) / this.stepHeaderWidth) - 1;
+    }
+
+    private handleDynamicHeaderChanges(visibleItemsNUmber: number) {
```

Comment:
Please, add the missing return type for functions in this file

Reply:
added